docs: add Armada scaling overview page (#266)
GameFabric/docs: src/multiplayer-servers/multiplayer-services/scaling.md

# Armada scaling

> GameFabric provides several mechanisms to control game server capacity per Region Type, balancing cost and player availability.

Reviewer: Can you add a glossary link to what a Region Type is?

Author: yes sure

> Each mechanism addresses a different aspect of scaling, from static configuration to demand-driven automation.

Reviewer 2: Avoid repeating `mechanism` especially since it's a vague term that doesn't mean much.

## Replicas and buffer size

Every Armada is configured with three core scaling parameters per Region Type:

- **Minimum Replicas**: the lowest number of game servers running at any time.
- **Maximum Replicas**: the upper bound on game servers that can be started.
- **Buffer Size**: the approximate number of game servers kept in the `Ready` state, waiting to be allocated.

These values determine how quickly players find a game server and how much idle capacity is maintained.
Getting them right depends on game server startup time, session duration, and expected concurrent users.

> For detailed guidance, including worked examples and input validation rules, see [Replicas and buffer size](/multiplayer-servers/multiplayer-services/armada-replicas-and-buffer).

Reviewer 2: Worked examples? Did you mean working examples?

> ## Dynamic buffer (Alpha)

Reviewer 2: What does Alpha mean here? Experimental? If so, I'd rather have a note in this section saying "use at your own risk / this is experimental" instead of it being part of the section name, especially since this could break linking when we remove the alpha marker

Author: got it from here: https://github.com/GameFabric/docs/blob/2684717c254ff873e84158f86b7dc28047731da7/src/multiplayer-servers/multiplayer-services/scale-to-zero.md?plain=1#L1

> Instead of choosing a fixed buffer size, the Dynamic Buffer option lets GameFabric adjust the buffer automatically based on observed player demand.

Reviewer: If you write `Dynamic buffer` in the title, you should be consistent with it in the text (`Dynamic Buffer`). I have no preference which one is correct, slightly opting for `Dynamic Buffer` since it's kind of a name, not just a noun.

When enabled, GameFabric recalculates the buffer at two levels: a baseline per Region Type derived from overall demand trends, and a per-Site local adjustment based on `Ready` and `Allocated` counts, startup time, and allocation patterns.
A slider controls the trade-off between cost efficiency and availability.

> Dynamic Buffer requires at least 24 hours of allocation data. After you enable it, it can take up to 48 hours to stabilize, so monitor it closely during that period.

Reviewer 2: ```suggestion
Dynamic Buffer requires at least 24 hours of allocation data. Once enabled, it takes up to 48 hours to stabilize, so monitor it closely during that period.
```
